src/components/User/Users/components/UserDetails/UserDetails.Request.vue

```diff
@@ -1,29 +1,43 @@
 <template>
   <div class="user-request">
-    <h3>Latest request</h3>
-    <p class="user-request__block text">
-      Create a {{ requestToReview.requestDetails.accountRoleToSet | roleIdToString | lowerCase }} account:
-      {{ requestToReview.state }}
-    </p>
-
-    <template v-if="isRequestPending">
-      <div class="app__form-actions user-details-request__form-actions">
-        <button
-          class="app__btn"
-          @click="approve"
-          :disabled="isPending"
-        >
-          Approve
-        </button>
+    <template v-if="requestToReview.state">
+      <h3>Latest request</h3>
+      <p class="user-request__block text">
+        Create a {{ requestToReview.requestDetails.accountRoleToSet | roleIdToString | lowerCase }} account:
+        {{ requestToReview.state }}
+      </p>
 
-        <button
-          class="app__btn app__btn--danger"
-          @click="showRejectModal"
-          :disabled="isPending"
-        >
-          Reject
-        </button>
-      </div>
+      <template v-if="isRequestPending">
+        <div class="app__form-actions user-details-request__form-actions">
+          <button
+            class="app__btn"
+            @click="approve"
+            :disabled="isPending"
+          >
+            Approve
+          </button>
+
+          <button
+            class="app__btn app__btn--danger"
+            @click="showRejectModal"
+            :disabled="isPending"
+          >
+            Reject
+          </button>
+        </div>
+      </template>
+    </template>
+
+    <template v-else-if="canRoleBeReset">
+      <button
+        class="app__btn
+               app__btn--danger
+               user-request__reset-to-unverified-btn"
+        @click="showRejectModal"
```

Comment:
make a separate modal for resetting. their UI is similar, but the business logic is completely different, we shouldn’t mess them

also your button showing logic gonna be easier

Reply:
Extracted